include/zephyr/math/dsp.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file zephyr/math/dsp.h
```

Comment:
As others and I pointed out earlier, can we settle on either `math` OR `dsp`, instead of using these two terms interchangeably on an arbitrary basis?

I agree with @ruuddw and, IMHO, we should settle for `dsp` since the CMSIS-DSP API, after which this subsystem is modelled, is a "DSP" library (and not just generic "math" library).

such that there are `include/zephyr/dsp/dsp.h`, `include/zephyr/dsp/basicmath.h`, ...

Reply:
Done